datafusion/__init__.py

```diff
@@ -190,7 +190,7 @@ def udaf(accum, input_type, return_type, state_type, volatility, name=None):
             "`accum` must implement the abstract base class Accumulator"
         )
     if name is None:
-        name = accum.__qualname__
+        name = accum.__qualname__.lower()
```

Comment:
Previously calling the UDAF in SQL would fail if `name` is not specified (i.e. both `MyAccumulator(b)` and `myaccumulator(b)` would fail in the example below if `name` is not specified). With this change both the uppercase & lowercase will work.